server/swap/swap_test.go

```diff
@@ -2003,18 +1998,54 @@ func TestMalformedSwap(t *testing.T) {
 	// Works with low fee, but now a confirmation.
 	tConfsSpoofer = 1
 	ensureNilErr(rig.sendSwap_maker(true))
-	rig.abcNode.invalidFeeRate = false
-	tConfsSpoofer = 0 // back to 0
+}
+
+func TestRetrySwapInit(t *testing.T) {
+	set := tPerfectLimitLimit(uint64(1e8), uint64(1e8), true)
+	matchInfo := set.matchInfos[0]
+	rig, cleanup := tNewTestRig(matchInfo)
+	defer cleanup()
+
+	rig.auth.swapReceived = make(chan struct{}, 1)
+
+	rig.swapper.Negotiate([]*order.MatchSet{set.matchSet})
+	ensureNilErr := makeEnsureNilErr(t)
+	ensureDupRequestErr := rpcErrorChecker(t, rig, msgjson.DuplicateRequestError)
+
+	ensureNilErr(rig.ackMatch_maker(true))
+	ensureNilErr(rig.ackMatch_taker(true))
+
+	ensureNilErr(rig.sendSwap_maker(true))
 
-	// Back to NewlyMatched
 	tracker := rig.getTracker()
+	// We're "rewinding time" back NewlyMatched status to be able to retry sending
+	// the same init request again.
+	tracker.mtx.Lock()
 	tracker.Status = order.NewlyMatched
-	// We're rewinding time to send the same init again. To support this hack,
-	// wait for swapSearching to clear.
-	time.Sleep(200 * time.Millisecond)
+	tracker.mtx.Unlock()
+
+	// We need to wait for swapSearching semaphore (it coordinates init request
+	// handling) to clear, so our retry request should eventually work (it has
+	// adequate fee and 0 confs).
+	startWaitingTime := time.Now()
+	for {
+		// We might get a couple of duplicate init request errors here, in that case
+		// we simply retry request later.
+		err := rig.sendSwap_maker(false)
+		if err == nil {
+			// We are done, finally got a retry attempt that worked.
+			ensureNilErr(rig.checkSwapStatusChange(order.MakerSwapCast, matchInfo.maker))
+			return
```

Comment:
Currently we are indeed done after this, but I'd prefer to `break` the loop instead and achieve the same outcome just by running to the end of the test function.  It'll allow any coded that may eventually get added after the `for` to work.  Linters will call that out, but `go` will still compile and run the test with unreachable code.